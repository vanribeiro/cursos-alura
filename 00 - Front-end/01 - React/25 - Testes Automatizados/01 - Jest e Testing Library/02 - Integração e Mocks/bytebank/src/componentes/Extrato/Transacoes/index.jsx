import React from 'react';

export default function Transacoes({ transacao, estilos }) {
  return (
    <li>
      <p className={estilos.mes}>{transacao.mes}</p>
      <div className={estilos.transacao}>
        <p data-testid="tipoTransacao">{transacao.transacao}</p>
        <span>{transacao.data}</span>
      </div>
      {transacao.transacao === 'TransferÃªncia' ? (
        <h3
          data-testid="valorTransacao"
          className={estilos.valor}
        >{`- ${transacao.valor.toLocaleString('pt-BR', { style: 'currency', currency: 'BRL' })}`}</h3>
      ) : (
        <h3
          data-testid="valorTransacao"
          className={estilos.valor}
        >{`${transacao.valor.toLocaleString('pt-BR', { style: 'currency', currency: 'BRL' })}`}</h3>
      )}
      <div className={estilos.divisor} />
    </li>
  );
}
