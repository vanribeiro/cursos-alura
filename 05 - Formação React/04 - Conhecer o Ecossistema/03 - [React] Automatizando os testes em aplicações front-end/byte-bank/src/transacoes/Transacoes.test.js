import { render } from "@testing-library/react";
import Transacoes from "./Transacoes";

describe('Componente que lista todas as transações do extrato', () => {
    it('O snapshot do component deve ser sempre o mesmo', () => {
        const infoTransacoes = [{
            id: 4, 
            transacao: "deposito", 
            valor: 50, 
            data:"19/02/2022"
        }];

        const { container } = render(<Transacoes transacoes={infoTransacoes} />);

        expect(container.firstChild).toMatchSnapshot()
    });
});