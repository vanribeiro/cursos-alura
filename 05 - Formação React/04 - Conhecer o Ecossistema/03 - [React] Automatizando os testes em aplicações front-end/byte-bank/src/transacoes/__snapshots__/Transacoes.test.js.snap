// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Componente que lista todas as transações do extrato O snapshot do component deve ser sempre o mesmo 1`] = `
<div
  data-testid="transacoes"
>
  <div
    class="transacao-container"
  >
    <p>
      19/02/2022
    </p>
    <p>
      deposito
    </p>
    <p>
      R$ 
      50
    </p>
  </div>
</div>
`;
